### ISSUE TITLE
Why bother implementing our own rwlatch

### ISSUE THREAD

**Issue by peileiscott** [pzhzqt/goostub#2]
Hi, pzhzqt. Thanks for this great project, I have a question about common/rwlatch.go, why do we have to implement our own rwlatch instead of using the reader-writer lock of golang directly.

**Comment by pzhzqt** [comment]
Good point! Actually I've been wondering the same thing since I noticed even bustub used both shared_mutex of cpp library and the rwlatch they defined themselves and I can't yet find a good reason for doing so. For now I would just leave it as it is and come back to it when I finished porting. If by then I still can't find a good reason I'll just remove it and use RWMutex directly.

I'll leave this issue open as a reminder. Thanks!
